docker/env/Dockerfile

```diff
@@ -1,18 +1,42 @@
-FROM python:3.8
+ARG PYTHON_IMAGE_TAG=3.8.3-slim-buster
 
-RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
-    apt-get install -y git openssl iproute2 sudo apt-transport-https \
-    ca-certificates curl gnupg2 software-properties-common rsync
+FROM python:$PYTHON_IMAGE_TAG as apt_base
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update
 
-# Install Docker cli only
-RUN curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key add - && \
-    DEBIAN_FRONTEND=noninteractive add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable" && \
-    DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -y docker-ce-cli  && \
-    sudo groupadd docker && \
-    sudo usermod -aG docker `whoami`
+# Add 3rd-party APT repositories.
+FROM apt_base as apt_repos
```

Comment:
Why doing this part in a separate build stage ? Seem a bit overkill, and you end up installing curl twice cause of that.

The three packages here are that big of difference in image size ?

Reply:
The cause that `software-properties-common` (and its friends) has a lot of dependencies which we don't need:

```
$ apt install --no-install-recommends --dry-run apt-transport-https gnupg2 software-properties-common
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  dirmngr distro-info-data gir1.2-glib-2.0 gir1.2-packagekitglib-1.0 gnupg gnupg-l10n gnupg-utils gpg gpg-agent gpg-wks-client gpg-wks-server gpgconf gpgsm iso-codes libapt-inst2.0 libassuan0 libcurl3-gnutls libdbus-1-3 libgirepository-1.0-1 libglib2.0-0
  libgssapi-krb5-2 libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0 libksba8 libldap-2.4-2 libldap-common libmpdec2 libnghttp2-14 libnpth0 libpackagekit-glib2-18 libpsl5 libpython3-stdlib libpython3.7-minimal libpython3.7-stdlib librtmp1 libsasl2-2
  libsasl2-modules-db libssh2-1 lsb-base lsb-release mime-support pinentry-curses python-apt-common python3 python3-apt python3-dbus python3-gi python3-minimal python3-pycurl python3-software-properties python3.7 python3.7-minimal
Suggested packages:
  dbus-user-session libpam-systemd pinentry-gnome3 tor parcimonie xloadimage scdaemon isoquery krb5-doc krb5-user lsb pinentry-doc python3-doc python3-tk python3-venv python3-apt-dbg python-apt-doc python-dbus-doc python3-dbus-dbg libcurl4-gnutls-dev python-pycurl-doc
  python3-pycurl-dbg python3.7-venv python3.7-doc binutils binfmt-support
Recommended packages:
  dbus libglib2.0-data shared-mime-info xdg-user-dirs krb5-locales packagekit publicsuffix libsasl2-modules bzip2 file xz-utils unattended-upgrades
The following NEW packages will be installed:
  apt-transport-https dirmngr distro-info-data gir1.2-glib-2.0 gir1.2-packagekitglib-1.0 gnupg gnupg-l10n gnupg-utils gnupg2 gpg gpg-agent gpg-wks-client gpg-wks-server gpgconf gpgsm iso-codes libapt-inst2.0 libassuan0 libcurl3-gnutls libdbus-1-3 libgirepository-1.0-1
  libglib2.0-0 libgssapi-krb5-2 libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0 libksba8 libldap-2.4-2 libldap-common libmpdec2 libnghttp2-14 libnpth0 libpackagekit-glib2-18 libpsl5 libpython3-stdlib libpython3.7-minimal libpython3.7-stdlib librtmp1 libsasl2-2
  libsasl2-modules-db libssh2-1 lsb-base lsb-release mime-support pinentry-curses python-apt-common python3 python3-apt python3-dbus python3-gi python3-minimal python3-pycurl python3-software-properties python3.7 python3.7-minimal software-properties-common
0 upgraded, 57 newly installed, 0 to remove and 0 not upgraded.
```

So, yes, these 3 images add about 70MB to the image.